tests/test_random.py

```diff
@@ -0,0 +1,62 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import unittest
+
+import retworkx
+
+
+class TestGNPRandomGraph(unittest.TestCase):
+
+    def test_random_gnp_directed(self):
+        graph = retworkx.directed_gnp_random_graph(20, .5, 10)
+        self.assertEqual(len(graph), 20)
+        self.assertEqual(len(graph.edges()), 104)
+
+    def test_random_gnp_directed_invalid_num_nodes(self):
+        with self.assertRaises(IndexError):
+            retworkx.directed_gnp_random_graph(-23, .5)
+
+    def test_random_gnp_directed_invalid_probability(self):
+        with self.assertRaises(IndexError):
+            retworkx.directed_gnp_random_graph(23, 123.5)
+
+    def test_random_gnp_undirected(self):
+        graph = retworkx.undirected_gnp_random_graph(20, .5, 10)
+        self.assertEqual(len(graph), 20)
+        self.assertEqual(len(graph.edges()), 105)
+
+    def test_random_gnp_undirected_invalid_num_nodes(self):
+        with self.assertRaises(IndexError):
+            retworkx.undirected_gnp_random_graph(-23, .5)
+
+    def test_random_gnp_undirected_invalid_probability(self):
+        with self.assertRaises(IndexError):
+            retworkx.undirected_gnp_random_graph(23, 123.5)
+
+    def test_random_gnp_directed_edges_close_to_probability(self):
+        edges = 0
```

Comment:
I'm not quite sure I understand this test.

The expected number of edges for a connection probability `p` and `n` nodes is `E = p n(n - 1) / 2`. Wouldn't it make sense to fix a random seed and check that the average number of edges is close to this expected value?

Reply:
I honestly just copied this from the networkx tests: https://github.com/networkx/networkx/blob/master/networkx/generators/tests/test_random_graphs.py#L248-L253 for the equivalent func over there. I thought it was validating that we got at least a reasonable number of edges averaged over 100 run when the probability was near 1 (~45 for graph). But looking at it more closely I'm confused what the point of these are too because it's not doing that.

Reply:
I just removed the tests in: https://github.com/Qiskit/retworkx/pull/94/commits/7bc46a07061cabf45925fb2d7d58441f4fd203f1 no point in keeping them around if it's not clear what we're actually verifying with the test. We can always add them back if we figure out the point of them.